src/Geocoder.php

```diff
@@ -8,216 +8,252 @@
-    public function autosuggest($input, $options=array())
-        {
-        $parameters = array("input"=>$input);
-
-        foreach ($options as $option)
-            $parameters = array_merge($parameters, $option);
-
-        return $this->performRequest("autosuggest", $parameters);
-        }
-
-
-    ////////////////////////////////////////////////////////////////
-
-    // Prepare the call to the API server
-    private function performRequest($command, $parameters)
-        {
-        // add the apikey to the parameters
-        $parameters["key"] = $this->api_key;
-
-        // make an array out of the dictionary so that each element is a key value pair glued together with an '=', and urlencode the parameters
-        $param_array = array();
-        foreach ($parameters as $key=>$value)
-            $param_array[] = "$key=" . urlencode($value);
-
-        // glue the array together unto a string with elements connected with '&'
-        $params = implode("&", $param_array);
-
-        // put the whole URL together now
-        $url = $this->base_url . $command . "?" . $params;
-
-        // cal the server
-        $json = $this->call($url);
-
-        $data = json_decode($json, true);
-
-        if (isset($data["error"]))
-          {
-          $this->error["code"]    = $data["error"]["code"];
-          $this->error["message"] = $data["error"]["message"];
-          $data = false;
-          }
-
-        return $data;
-        }
-
-
-    // Make the call to the API server
-    private function call($url)
-        {
-        $handle = curl_init();
-
-        // set the options
-        curl_setopt($handle, CURLOPT_URL, $url);
-        curl_setopt($handle, CURLOPT_HTTPHEADER, array("X-W3W-Wrapper: " . $this->header));
-        curl_setopt($handle, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($handle, CURLOPT_ENCODING, "");
-        curl_setopt($handle, CURLOPT_MAXREDIRS, 10);
-        curl_setopt($handle, CURLOPT_TIMEOUT, 30);
-        curl_setopt($handle, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1);
-        curl_setopt($handle, CURLOPT_CUSTOMREQUEST, "GET");
-
-        // make the call
-        $output = curl_exec($handle);
-        if ($output == false)
-            {
-            $this->error["code"]    = "BadConnection";
-            $this->error["message"] = curl_error($handle);
-            }
-
-        curl_close($handle);
-
-        // return the json
-        return $output;
-        }
+class Geocoder
+{
+
+  private $version = "3.4.0";  // if changing this, remember to change the comment block at the top, and match everything with the git tag
+  private $apiKey = "";
+  private $error = false;
+  private $baseUrl = 'https://api.what3words.com/v3/';
```

Comment:
I reckon the camelCase prop name changes would technically make this a breaking change? i.e. `version=4.0.0`

Reply:
these prop names were only ever used privately, specifically with apiKey and baseUrl. I'm unsure if there's a use case wherein anyone would use `$geocoder->baseUrl` or `$geocoder->apiKey` from outside the Geocoder class. Unless we're exposing these properties publicly, I doubt we would for api key.

Reply:
yeah my concern was that this is a public package so we don't really know what it's downstream usage looks like. Also not sure how php handles package dep management i.e. are versions pinned by default. Happy to approve, good enough to sign post and keep an eye post release on bd queries

Reply:
Sounds good. Yeah, a very valid concern, this is to be published to packagist from what I know, how the package dep works, is also beyond me.